sdk/storage/src/clients/container_client.rs

```diff
@@ -0,0 +1,56 @@
+use crate::clients::StorageClient;
+use azure_core::errors::AzureError;
+use http::method::Method;
+use http::request::{Builder, Request};
+use std::sync::Arc;
+
+pub trait AsContainerClient<CN: Into<String>> {
```

Comment:
What is this trait needed for?

Reply:
The idea is to keep the cosmos approach by having the fluid syntax like:

1. Create an Azure storage account client (by using for example the SAS token).
2. Call `as_storage_client` and it becomes a storage client (which for example exposes the `list_containers` method).
3. Call `as_container_client` specifying a container name as it becomes a container client  (which for example exposes the `list_blobs` method).
4. etc...

This works well but I wanted to avoid resorting to borrow (not again! 🗡️🗡️) or deep cloning so I've tried the `Arc` route. It seems a good balance but I needed a trait that I could implement for `Arc<Box<...>>`.  It's transparent for the end user (the just call `as_container_client` without even noticing the trait) but it's a mild nuisance for the maintainers.

Reply:
Why are we wrapping `ContainerClient` in an `Arc<Box<>>`? Why can't `as_container_client` just return a `ContainerClient`? `ContainerClient` already contains an `Arc` so cloning should be relatively cheap. We do clone the `container_name` `String` but that should be relatively cheap. If it becomes a problem we can wrap that in the `Arc` too.

Reply:
In other words, I'd be happy if clients were all cheaply cloneable without having to wrap them in `Arc`s.